flink-runtime/src/test/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolderTest.java

```diff
@@ -442,6 +442,25 @@ public void testCheckpointFailsIfSendingEventFailedBeforeTrigger() throws Except
         assertTrue(checkpointResult.isCompletedExceptionally());
     }
 
+    @Test
+    public void testFailJobWhenExceptionThrownFromStartingOperatorCoordinatorHolder()
+            throws Exception {
+        Function<OperatorCoordinator.Context, OperatorCoordinator> coordinatorProvider =
+                context ->
+                        new TestingOperatorCoordinator(context) {
+                            @Override
+                            public void start() {
+                                throw new RuntimeException("Artificial Exception");
+                            }
+                        };
+        final EventReceivingTasks tasks = EventReceivingTasks.createForRunningTasks();
+        final OperatorCoordinatorHolder holder =
+                createCoordinatorHolder(tasks, coordinatorProvider);
+
+        assertNotNull(globalFailure);
```

Comment:
Can you add an assertion to check that the job is cancelled with the appropriate error?